src/io.jl

```diff
@@ -85,32 +88,51 @@ function read_cellmap(habitat_file::String, is_res::Bool, ::Type{T}) where {T}
     gmap, rastermeta
 end
 
-function _ascii_grid_reader(T, file)
-    f = endswith(file, ".gz") ? GZip.open(file, "r") : open(file, "r")
-    rastermeta = _ascii_grid_read_header(file, f)
-    c = Matrix{T}(undef,0,0)
-    ss = 6
-    if rastermeta.nodata == -Inf
-        ss = 5
-    end
-    try
-        c = readdlm(f, T; skipstart = ss)
-    catch
-        seek(f, 0)
+function _grid_reader(T, file) # make a wrapper around read_raster to return rastermeta as well
+    #=if endswith(file, ".gz") ?
+        f = GZip.open(file, "r")
+        rastermeta = _ascii_grid_read_header(file, f)
+        c = Matrix{T}(undef,0,0)
+        ss = 6
+        if rastermeta.nodata == -Inf
+            ss = 5
+        end
         try
-            d = readdlm(f; skipstart = ss)
-            d = d[:, 1:end-1]
-            c = map(T, d)
+            c = readdlm(f, T; skipstart = ss)
         catch
-            error("Failed to read habitat map. There may be errors in your file.")
+            seek(f, 0)
+            try
+                d = readdlm(f; skipstart = ss)
+                d = d[:, 1:end-1]
+                c = map(T, d)
+            catch
+                error("Failed to read habitat map. There may be errors in your file.")
+            end
         end
-    end
-    map!(x -> x == rastermeta.nodata ? -9999. : x , c, c)
-    close(f)
-    c, rastermeta
+        map!(x -> x == rastermeta.nodata ? -9999. : x , c, c)
+        close(f)
+        return c, rastermeta
+    else=#
+    c, wkt, transform = read_raster(file, T)
+
+    rastermeta = get_raster_meta(c, wkt, transform)
+
+    return c, rastermeta
+
+end
+
+function get_raster_meta(habitat_file, wkt, transform)
+    dims = size(habitat_file)
+    ncols = dims[2]
+    nrows = dims[1]
+    xllcorner = transform[1]
+    yllcorner = nrows - transform[4]
+    cellsize = transform[2]
+    nodata = -9999 # set in read_raster (overwrites old nodata val)
+    RasterMeta(ncols, nrows, xllcorner, yllcorner, cellsize, nodata, transform, wkt)
 end
 
-function _ascii_grid_read_header(habitat_file, f)
```

Comment:
This function should be deleted, but kept for now pending any need for it for GZip stuff.